### PR TITLE
Support json outputs

### PR DESCRIPTION
**Pull request by chmp**
IPython allows to add arbitrary JSON data as output. This pull request adds support to runipy for this output type. It contains corresponding notebooks in the test directories to illustrate the changes.

### CODE REVIEW

**Comment by jakirkham** [comment]
Interesting idea. Are you planning on using this somehow?

File: runipy/notebook_runner.py [hunk 1]
```diff
@@ -194,8 +198,16 @@ def run_cell(self, cell):
                         raise NotImplementedError(
                             'unhandled mime type: %s' % mime
                         )
-
-                    setattr(out, attr, data)
+                    
+                    # In notebook version <= 3 JSON data is stored as a string
+                    # Evaluation of IPython2's JSON gives strings directly
+                    # Therefore do not encode for IPython versions prior to 3
+                    json_encode = (
```
Comment (jakirkham) [inline runipy/notebook_runner.py]: Interesting idea. Are you planning on using this somehow?

Comment (chmp) [inline runipy/notebook_runner.py]: Yes, I'm currently experimenting with building a reporting tool based on IPython notebooks. JSON outputs are used to generate machine-readable meta-data.

**Comment by Juanlu001** [comment]
Another use case: when using the extension [version_information](https://github.com/jrjohansson/version_information), the data is exported to the notebook in a handful of formats, including `application/json`, like this:

```
[...]
 "execution_count": 2,
 "metadata": {
  "collapsed": false
 },
 "outputs": [
  {
   "data": {
    "application/json": {
     "Software versions": [
      {
       "module": "Python",
       "version": "2.7.11 64bit [MSC v.1500 64 bit (AMD64)]"
      },
      {
       "module": "IPython",
       "version": "4.2.0"
      },
      {
       "module": "OS",
       "version": "Windows 7 6.1.7601 SP1"
      }
     ]
    },
    "text/html": [
     "<table><tr><th>Software</th><th>Version</th></tr><tr><td>Python</td><td>2.7.11 64bit [MSC v.1500 64 bit (AMD64)]</td></tr><tr><td>IPython</td><td>4.2.0</td></tr><tr><td>OS</td><td>Windows 7 6.1.7601 SP1</td></tr><tr><td colspan='2'>Wed May 11 17:19:41 2016 Romance Daylight Time</td></tr></table>"
    ],
    "text/latex": [
     "\\begin{tabular}{|l|l|}\\hline\n",
     "{\\bf Software} & {\\bf Version} \\\\ \\hline\\hline\n",
     "Python & 2.7.11 64bit [MSC v.1500 64 bit (AMD64)] \\\\ \\hline\n",
     "IPython & 4.2.0 \\\\ \\hline\n",
     "OS & Windows 7 6.1.7601 SP1 \\\\ \\hline\n",
     "\\hline \\multicolumn{2}{|l|}{Wed May 11 17:19:41 2016 Romance Daylight Time} \\\\ \\hline\n",
     "\\end{tabular}\n"
    ],
    "text/plain": [
     "Software versions\n",
     "Python 2.7.11 64bit [MSC v.1500 64 bit (AMD64)]\n",
     "IPython 4.2.0\n",
     "OS Windows 7 6.1.7601 SP1\n",
     "Wed May 11 17:19:41 2016 Romance Daylight Time"
    ]
   },
   "execution_count": 2,
   "metadata": {},
   "output_type": "execute_result"
  }
 ],
 "source": [
  "%version_information"
 ]
},
[...]
```

With the current version of runipy one cannot use these kind of extensions.

**Comment by jakirkham** [comment]
Thanks @chmp.

**Comment by chmp** [comment]
Thanks for merging.

**Pull request merged**